### PR TITLE
[runtime] Enhancement: Refine timer resolution

### PR DESCRIPTION
**Pull request by iyzhang**
This PR changes the timer resolution/number of iterations to run the scheduler before checking the clock depending on the remaining time out left.

### CODE REVIEW

File: tests/rust/common/libos.rs [hunk 1]
```diff
@@ -112,17 +112,26 @@ impl DummyLibOS {
         // Otherwise, actually run the scheduler.
         // Put the QToken into a single element array.
         let qt_array: [QToken; 1] = [qt];
-        let start: Instant = Instant::now();
+        let mut prev: Instant = Instant::now();
+        let mut remaining_time: Duration = timeout.unwrap_or(DEFAULT_TIMEOUT);
 
         // Call run_any() until the task finishes.
-        while Instant::now() <= start + timeout.unwrap_or(DEFAULT_TIMEOUT) {
+        // Call run_any() until the task finishes.
```
Comment (ppenna) [inline tests/rust/common/libos.rs:119]: Duplicate comment.

File: src/rust/runtime/mod.rs [hunk 2]
```diff
@@ -344,8 +356,12 @@ impl SharedDemiRuntime {
 
     /// Runs the scheduler for one [TIMER_RESOLUTION] quanta, returning any ready task. Importantly does not modify
     /// the clock.
-    fn run_next(&mut self) -> Option<(QToken, QDesc, OperationResult)> {
-        if let Some(boxed_task) = self.scheduler.get_next_completed_task(TIMER_RESOLUTION) {
+    fn run_next(&mut self, timeout: Duration) -> Option<(QToken, QDesc, OperationResult)> {
+        let iterations: usize = match timeout {
+            timeout if timeout.as_secs() > 0 => TIMER_RESOLUTION,
+            _ => 16,
```
Comment (ppenna) [inline src/rust/runtime/mod.rs:362]: Could we have this in a file-scope constant?

File: src/rust/runtime/mod.rs [hunk 3]
```diff
@@ -344,8 +356,12 @@ impl SharedDemiRuntime {
 
     /// Runs the scheduler for one [TIMER_RESOLUTION] quanta, returning any ready task. Importantly does not modify
     /// the clock.
-    fn run_next(&mut self) -> Option<(QToken, QDesc, OperationResult)> {
-        if let Some(boxed_task) = self.scheduler.get_next_completed_task(TIMER_RESOLUTION) {
+    fn run_next(&mut self, timeout: Duration) -> Option<(QToken, QDesc, OperationResult)> {
+        let iterations: usize = match timeout {
+            timeout if timeout.as_secs() > 0 => TIMER_RESOLUTION,
```
Comment (ppenna) [inline src/rust/runtime/mod.rs:361]: Could we have this in a file-scope constant?

File: tests/rust/common/libos.rs (hunk 1 above)
Comment (iyzhang) [reply tests/rust/common/libos.rs]: Got it

**Review by ppenna — APPROVED** [review]
LGTM

**Review by ppenna — APPROVED** [review]

**Pull request merged**